cpp/include/cuvs/preprocessing/spectral_embedding.hpp

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <raft/core/device_coo_matrix.hpp>
+#include <raft/core/device_csr_matrix.hpp>
+#include <raft/core/device_mdspan.hpp>
+#include <raft/core/resources.hpp>
+
+namespace cuvs::preprocessing::spectral_embedding {
+
+/**
+ * @brief Parameters for spectral embedding algorithm
+ */
+struct params {
+  /** @brief The number of components to reduce the data to. */
+  int n_components;
+
+  /** @brief The number of neighbors to use for the nearest neighbors graph. */
+  int n_neighbors;
+
+  /** @brief Whether to normalize the Laplacian matrix. */
+  bool norm_laplacian;
+
+  /** @brief Whether to drop the first eigenvector. */
+  bool drop_first;
+
+  /** @brief Random seed for reproducibility */
+  uint64_t seed;
+};
+
+void create_connectivity_graph(raft::resources const& handle,
```

Comment:
Is there a reason this needs to be exposed through the public APIs? This is all implementation details and should only be used within `src`/

Reply:
To be clear- the user should only have to call `build()` and `tranform()` functions from cuVS to build a spectral embedding and then transform a series of vectors into that space. If we don't have those separated yet, we can just do a `transform` for now, but that should really be the only function we need to expose.